src/main/java/roomescape/reservation/domain/Reservation.java

```diff
@@ -0,0 +1,26 @@
+package roomescape.reservation.domain;
+
+import java.time.LocalDate;
+
+import roomescape.theme.domain.Theme;
+import roomescape.time.domain.ReservationTime;
+
+public record Reservation(long id, String name, LocalDate date, ReservationTime time, Theme theme) {
+    private static final long UNDEFINED = 0;
+
+    public Reservation(String name, LocalDate date, ReservationTime time, Theme theme) {
+        this(UNDEFINED, name, date, time, theme);
+        validateName(name);
+    }
```

Comment:
생성자 체이닝 활용 👍 
다만 이런 경우 id 까지 주입 받는 생성자는 검증 로직을 태우기 어려울텐데요,
아래와 같이 사용해보는 건 어떨까요?

```suggestion
public record Reservation(long id, String name, LocalDate date, ReservationTime time, Theme theme) {
    private static final long UNDEFINED = 0;

    public Reservation {
        validateName(name);
    }

    public Reservation(String name, LocalDate date, ReservationTime time, Theme theme) {
        this(UNDEFINED, name, date, time, theme);
    }
```